### PR TITLE
Enabled Garbage Collection for Flux

### PR DESCRIPTION
**Pull request by yradsmikham**
used to remove resources in rings implementation

### CODE REVIEW

File: cluster/azure/aks-gitops/variables.tf [hunk 1]
```diff
@@ -11,6 +11,10 @@ variable "acr_enabled" {
   type = "string"
 }
 
+variable "gc_enabled" {
+  type = "string"
```
Comment (evanlouie) [inline cluster/azure/aks-gitops/variables.tf]: should use boolean instead?

Comment (yradsmikham) [reply cluster/azure/aks-gitops/variables.tf]: Source: https://www.terraform.io/docs/configuration/expressions.html

> _Terraform automatically converts number and bool values to strings when needed. It also converts strings to numbers or bools, as long as the string contains a valid representation of a number or bool value.
> 
> true converts to "true", and vice-versa
> false converts to "false", and vice-versa_

Comment (yradsmikham) [reply cluster/azure/aks-gitops/variables.tf]: I left it as string just to be consistent with the other "boolean" variable types, but I'm happy to change it. Regardless, Terraform is intelligent enough to treat variable types appropriately.

**Comment by andrebriggs** [comment]
Closes #228

**Pull request merged**